c/src/ml-api-service-extension.c

```diff
@@ -829,14 +835,89 @@ ml_service_extension_destroy_internal (ml_service_s * mls)
   return ML_ERROR_NONE;
 }
 
+/**
+ * @brief Internal function to parse service info from config file.
+ */
+static int
+_ml_service_remote_conf_to_opt (JsonObject * object, const gchar * name,
+    ml_option_h option)
+{
+  int status;
+  JsonObject *remote_service_object;
+  const gchar *val = NULL;
+  GList *list = NULL, *iter;
+
+  remote_service_object = json_object_get_object_member (object, name);
+
+  list = json_object_get_members (remote_service_object);
+  for (iter = list; iter != NULL; iter = g_list_next (iter)) {
+    val = json_object_get_string_member (remote_service_object, iter->data);
+    status = ml_option_set (option, iter->data, g_strdup (val), g_free);
+    if (status != ML_ERROR_NONE) {
+      _ml_error_report ("Failed to set %s option: %s.", (gchar *) list->data,
+          val);
+      option = NULL;
+      break;
+    }
+  }
+  g_list_free (list);
+
+  return status;
+}
+
+/**
+ * @brief Internal function to parse configuration file for remote service.
+ */
+static int
+_ml_service_remote_conf_parse_json (ml_service_s * mls, JsonObject * object)
+{
+  int status;
+  ml_option_h option;
+  gchar *node_type = NULL;
+
+  if (json_object_has_member (object, "remote-sender")) {
+    node_type = g_strdup ("remote-sender");
+  } else if (json_object_has_member (object, "remote-receiver")) {
+    node_type = g_strdup ("remote-receiver");
+  } else {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "Not supported ml-service-remote node type, cannot get the valid type from configuration.");
+  }
+
+  status = ml_option_create (&option);
+  if (status != ML_ERROR_NONE) {
+    _ml_error_report_return (status, "Failed to create ml-option.");
+  }
+
+  status = ml_option_set (option, "node-type", node_type, g_free);
+  if (status != ML_ERROR_NONE) {
+    ml_option_destroy (option);
+    _ml_error_report_return (status,
+        "Cannot parse config file. Failed to set node type");
+  }
+
+  status = _ml_service_remote_conf_to_opt (object, node_type, option);
+  if (status != ML_ERROR_NONE) {
+    ml_option_destroy (option);
```

Comment:
If the condition is true, option becomes NULL.  so option cannot be destroyed.

Reply:
Good point. Updated.